sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/implementation/ChangeFeedProcessorBuilderImpl.java

```diff
@@ -308,16 +312,26 @@ public ChangeFeedProcessorBuilderImpl withCollectionResourceId(String collection
     /**
      * Sets an existing {@link CosmosAsyncContainer} to be used to read from the leases collection.
      *
-     * @param leaseDocumentClient the instance of {@link CosmosAsyncContainer} to use.
+     * @param leaseClient the instance of {@link CosmosAsyncContainer} to use.
      * @return current Builder.
      */
     @Override
-    public ChangeFeedProcessorBuilderImpl leaseContainer(CosmosAsyncContainer leaseDocumentClient) {
-        if (leaseDocumentClient == null) {
-            throw new IllegalArgumentException("leaseContextClient");
+    public ChangeFeedProcessorBuilderImpl leaseContainer(CosmosAsyncContainer leaseClient) {
+        if (leaseClient == null) {
+            throw new IllegalArgumentException("leaseClient");
+        }
+
+        if (!getContextClient(leaseClient).isContentResponseOnWriteEnabled()) {
+            throw new IllegalArgumentException("leaseClient: content response on write setting must be enabled");
+        }
+
+        ConsistencyLevel consistencyLevel = getContextClient(leaseClient).getConsistencyLevel();
+        if (consistencyLevel == ConsistencyLevel.CONSISTENT_PREFIX || consistencyLevel == ConsistencyLevel.EVENTUAL) {
+            logger.warn("leaseClient consistency level setting are less then expected which is SESSION");
```

Comment:
I think we should fail on these cases, logs may go unnoticed by user

Reply:
The CFP should continue to work with lesser than SESSION consistency level; the downside of it is that lease documents might not be updated in a timely fashion by the current CFP instance in certain conditions which at worst can lead to documents being seen more than once. SESSION or better will help avoid that because once the lease document is written, the CFP instance will have it right away in most scenarios across all its workers.